### ISSUE TITLE
Tab spawned from private tab back history not always private.

### ISSUE THREAD

**Issue by windwakr** [Infocatcher/Private_Tab#230]
If I right click the back arrow in a private tab, then middle click one of the pages to open on in a new tab, sometimes it will open in a non private tab.

Tested using Private Tab 0.1.9.1 and Firefox 45 ESR with a new profile.

**Comment by windwakr** [comment]
Just noticed #131, which is somewhat similar to this. I have enabled debugging and caught what it says when this happens:
```
[Private Tab] 19:13:609 Tab opened: New Tab (empty)
Inherit private state: undefined
[Private Tab] 19:13:617 Selected tab not yet loaded, wait
[Private Tab] 19:13:644 updateWindowTitle() Mozilla Firefox
[Private Tab] 19:13:667 updateTabsInTitlebar() => TabsInTitlebar._update(true)
[Private Tab] 19:13:668 Hide tracking protection icon
[Private Tab] 19:13:668 setWindowBusy(): true
[Private Tab] 19:13:670 setWindowBusy(): false
[Private Tab] 19:13:691 Tab restored, private attribute: false, label: GitHub - Infocatcher/Private_Tab: Adds private tabs, restartless extension for Firefox (20.0+) and SeaMonkey (2.17+)
[Private Tab] 19:13:745 updateDownloadPanel() => reinitialize download indicator
```

And this is what I get logged when it works properly and opens a private tab:
```
[Private Tab] 19:41:438 Tab opened: New Tab (empty)
Inherit private state: undefined
[Private Tab] 19:41:446 Selected tab not yet loaded, wait
[Private Tab] 19:41:466 updateWindowTitle() Mozilla Firefox
[Private Tab] 19:41:497 updateTabsInTitlebar() => TabsInTitlebar._update(true)
[Private Tab] 19:41:497 Hide tracking protection icon
[Private Tab] 19:41:498 setWindowBusy(): true
[Private Tab] 19:41:499 setWindowBusy(): false
[Private Tab] 19:41:520 Tab restored, private attribute: true, label: Search · private tab · GitHub
[Private Tab] 19:41:520 Make restored tab private
Only internal code is allowed to set the usePrivateBrowsing attribute bootstrap.js:3557:2
[Private Tab] 19:41:520 Set usePrivateBrowsing to true
Tab: Search · private tab · GitHub
[Private Tab] 19:41:529 PrivateTab:PrivateChanged + tab is selected => updateWindowTitle()
[Private Tab] 19:41:529 updateWindowTitle() Mozilla Firefox (Private Browsing)
[Private Tab] 19:41:549 updateTabsInTitlebar() => TabsInTitlebar._update(true)
[Private Tab] 19:41:637 updateDownloadPanel() => reinitialize download indicator
[Private Tab] 19:41:860 [patcher] gBrowser.setIcon(): isPrivate = true
[Private Tab] 19:41:861 PrivateBrowsingUtils.isWindowPrivate(): override to true
```

**Comment by Infocatcher** [comment]
> Tab restored, private attribute: false

May be this is something with sessions mechanism... Looks like sometimes tab state isn't updated yet.

But also this may be fixed from "Tab opened: New Tab <ins>(empty)</ins>" side.
And following fix should help: f08cc143c976e98ba8e1ff55bc2105d104a40ea1

Test version: <a href="http://infocatcher.ucoz.net/temp/pt/private_tab-0.1.9.2pre5-fx-sm.xpi">private_tab-0.1.9.2pre5-fx-sm.xpi</a> (<a href="https://github.com/Infocatcher/Private_Tab/tree/f08cc143c976e98ba8e1ff55bc2105d104a40ea1">source code</a>).

**Comment by windwakr** [comment]
Yeah, that seems to fix it.
